workflow-templates/im-run-tf-destroy.yml

```
@@ -12,86 +12,402 @@
 #    - Ensure each of the repo-level and env-level secrets used in this workflow have been populated by an admin in your repository.
 
 name: Terraform Destroy
-run-name: Terraform Destroy ${{ inputs.environment }}
+run-name: Terraform destroy ${{ inputs.root-module }} at ${{ inputs.branch-tag-sha }}
 on:
   workflow_dispatch:
     inputs:
-      environment:
-        description: The environment the destroy should be run against.
+      # This is required because a tf init has to be performed before the terraform destroy command
+      branch-tag-sha:
+        description: The branch, tag or sha of the terraform that has the configuration of the resources that will be destroyed.
+        required: true
+      root-module:
```

```suggestion
      root-module: # This is used in loading environments secrets and was previously called environment
```

I think we need a little bit of a description on this new input.

---

Ok I think it is fine.  Posibbly seeing dev, qa, stage etc in the dropdown is intuitive enough.